src/CacheTagsInvalidator.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\cloudflare\CacheTagsInvalidator.
+ */
+
+namespace Drupal\cloudflare;
+
+use CloudFlarePhpSdk\ApiEndpoints\ZoneApi;
+use CloudFlarePhpSdk\Exceptions\CloudFlareHttpException;
+use CloudFlarePhpSdk\Exceptions\CloudFlareApiException;
+use Drupal\cloudflare\EventSubscriber\CloudFlareCacheTagHeaderGenerator;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Cache\CacheTagsInvalidatorInterface;
+
+/**
+ * Cache tags invalidator implementation that invalidates CloudFlare.
+ */
+class CacheTagsInvalidator implements CacheTagsInvalidatorInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function invalidateTags(array $tags) {
+    // When either an extension (module/theme) is (un)installed, purge
+    // everything.
+    if (in_array('config:core.extension', $tags)) {
+      // @todo Purge everything. Blocked on https://github.com/d8-contrib-modules/cloudflare/issues/16.
+      return;
+    }
+
+    // Also invalidate the cache tags as hashes, to automatically also work for
+    // responses that exceed CloudFlare's Cache-Tag header limit.
+    $hashes = CloudFlareCacheTagHeaderGenerator::cacheTagsToHashes($tags);
```

Comment:
Is this line necessary.  Doesn't the onResponse inside CloudFlareCacheTagHeaderGenerator already accomplish this?